src/rashdf/geom.py

```diff
@@ -488,3 +574,171 @@ def ineffective_points(self) -> GeoDataFrame:  # noqa D102
 
     def blocked_obstructions(self) -> GeoDataFrame:  # noqa D102
         raise NotImplementedError
+
+    def steady_flow_names(self) -> list:
+        """Return the profile information for each steady flow event.
+
+        Returns
+        -------
+        DataFrame
+            A Dataframe containing the profile names for each event
+        """
+        path = "Results/Steady/Output/Output Blocks/Base Output/Steady Profiles"
+        if path not in self:
+            return pd.DataFrame()
+
+        profile_data = self[path]
+        profile_attrs = profile_data["Profile Names"][()]
+        profilenames = [x.decode("utf-8") for x in profile_attrs]
+
+        return profilenames
+
+    def cross_sections_wsel(self) -> pd.DataFrame:
+        """Return the WSEL information for each 1D Cross Section.
+
+        Returns
+        -------
+        DataFrame
+            A Dataframe containing the water surface elevations for each cross section and event
+        """
+        path = "Results/Steady/Output/Output Blocks/Base Output/Steady Profiles/Cross Sections"
+        if path not in self:
+            return pd.DataFrame()
+
+        wsel_data = self[path]
+        wsel_attrs = wsel_data["Water Surface"][()]
+
+        xs_df = self.cross_sections()
+        xs_cols = ["River", "Reach", "RS"]
+        steadyprofiles = self.steady_flow_names()
+
+        wsel_df = pd.DataFrame(wsel_attrs, index=steadyprofiles)
+        wsel_df_t = wsel_df.transpose().copy()
+        for col in xs_cols:
+            wsel_df_t[col] = xs_df[col]
+        # wsel_df_t.set_index(xs_cols, inplace=True)
+
+        return wsel_df_t
+
+    def cross_sections_elevations(self) -> pd.DataFrame:
+        """Return the model cross section elevation information.
+
+        Returns
+        -------
+        DataFrame
+            A DataFrame containing the model cross section elevation information if they exist.
+        """
+        path = "/Geometry/Cross Sections"
+        if path not in self:
+            return pd.DataFrame()
+
+        xselev_data = self[path]
+        xs_df = self.cross_sections()
+        elevations = list()
+        for part_start, part_cnt in xselev_data["Station Elevation Info"][()]:
+            xzdata = xselev_data["Station Elevation Values"][()][
+                part_start : part_start + part_cnt
+            ]
+            elevations.append(xzdata)
+
+        xs_elev_df = xs_df[
+            ["xs_id", "River", "Reach", "RS", "Left Bank", "Right Bank"]
+        ].copy()
+        xs_elev_df["Left Bank"] = xs_elev_df["Left Bank"].round(2).astype(str)
+        xs_elev_df["Right Bank"] = xs_elev_df["Right Bank"].round(2).astype(str)
```

Comment:
Re: `.round(2).astype(str)` -- are there issues occurring if bank stations aren't rounded to 2 places and converted to string?

Reply:
There are some instances (not always, but sometimes) where the floats in the hdf are precise in the banks and imprecise in the elevation data and vice versa. Most cases models only use 2 decimal places.

I like the round_to variable though for people to adjust if needed. Applying it in the xs elevations function as well